test/unit/mock-http.test.ts

```diff
@@ -92,39 +94,92 @@ describe('Mocked API tests', () => {
 
     nock('http://localhost')
       .post(createAssemblyRegex)
-      .reply(400, { error: 'INVALID_FILE_META_DATA' })
+      .reply(400, { error: 'INVALID_FILE_META_DATA', message: 'Invalid file metadata' })
 
     await expect(client.createAssembly()).rejects.toThrow(
       expect.objectContaining({
-        transloaditErrorCode: 'INVALID_FILE_META_DATA',
-        cause: {
+        response: {
           error: 'INVALID_FILE_META_DATA',
+          message: 'Invalid file metadata',
         },
-        message: 'Response code 400 (Bad Request) INVALID_FILE_META_DATA',
+        message: 'API error (HTTP 400) INVALID_FILE_META_DATA: Invalid file metadata',
       })
     )
   })
 
-  it('should return assemblyId and response.body in Error', async () => {
+  it('should return informative errors', async () => {
     const client = getLocalClient()
 
     nock('http://localhost').post(createAssemblyRegex).reply(400, {
       error: 'INVALID_FILE_META_DATA',
+      message: 'Invalid file metadata',
       assembly_id: '123',
       assembly_ssl_url: 'https://api2-oltu.transloadit.com/assemblies/foo',
     })
 
-    await expect(client.createAssembly()).rejects.toThrow(
+    const promise = client.createAssembly()
+    await expect(promise).rejects.toThrow(
       expect.objectContaining({
-        assemblyId: '123',
         message:
-          'Response code 400 (Bad Request) INVALID_FILE_META_DATA - https://api2-oltu.transloadit.com/assemblies/foo',
-        response: expect.objectContaining({
-          body: expect.objectContaining({ assembly_id: '123' }),
-        }),
-        cause: expect.objectContaining({ assembly_id: '123' }),
+          'API error (HTTP 400) INVALID_FILE_META_DATA: Invalid file metadata https://api2-oltu.transloadit.com/assemblies/foo',
+        response: expect.objectContaining({ assembly_id: '123' }),
       })
     )
+
+    try {
+      await promise
+    } catch (err) {
+      expect(inspect(err).split('\n')).toEqual([
```

Comment:
Nice, this is a clever way to assert multiline strings that don’t have to match exactly.